base/testfiles-ltcmd/ltcmd008.lvt

```diff
@@ -0,0 +1,66 @@
+
+\documentclass{minimal}
+\input{regression-test}
+\RequirePackage[check-declarations]{expl3}
+\ExplSyntaxOn
+\debug_on:n { deprecation }
+\ExplSyntaxOff
+
+\START
+\AUTHOR{Joseph Wright}
+
+\ExplSyntaxOn
+
+\TEST { Basic~definitions~using~={...} }
+  {
+    \DeclareDocumentCommand \foo { = { TOC-entry } O{#2} m } { \TYPE { \detokenize { (#1) (#2) } } }
+    \foo { bar }
+    \foo [ bar ] { baz }
+  }
+
+\TEST { Collecting~keyvals }
+  {
+    \DeclareDocumentCommand \foo { = { TOC-entry } O{ } m } { \TYPE { \detokenize { (#1) (#2) } } }
+    \foo [ bong ] { baz }
+    \foo [ bong = bang ] { baz }
+    \foo [ = , bong ] { baz }
+    \foo [ { bong =  bang } ] { baz }
```

Comment:
I'm not sure about the expected outcome of this test.  Are the braces supposed to hide the keyval as a free text argument with an `=`?  Currently it's seeing `\foo [ { bong =  bang } ] { baz }` exactly like `\foo [ bong = bang ] { baz }` because the `D` argument grabber removes a set of braces from within the delimiters.

Reply:
The brace stripping is supposed to be suppressed for these cases (which is why this is more than a processor), which means that the expected result is `TOC-entry={bong=bang}`. That _was_ the case, but I see it changed in 11cc8bd3706d8241065386ae10531f9d75437025 ...

Reply:
... but as you adjusted other stuff after that, it will take me a while to track down what's gone wrong. If I have a chance I will do.

Reply:
`\foo [ { bong = bang } ] { baz }`  is supposed to turn the argument in a free text one, just like `\item[{foo [x] bar}]` in  classic LaTeX hides the  `]` in the item text so that it is not mistaken for the end of the optional argument. 
This is the std way people do this therefore has to be maintained. We can't go and say, fix your document and put the braces inside.

Reply:
I see the issue: in setting up the two aspects (brace strip suppression and post-processing to hunt for `=`), I've messed up and the brace stripping suppression 'gets lost' in the internals. I'll work out a fix shortly unless @PhelypeOleinik  beat me to it: the problem is in `\@@_prepare_signature:n` or perhaps what is done subsequently.

Reply:
> That _was_ the case, but I see it changed in [11cc8bd](https://github.com/latex3/latex2e/commit/11cc8bd3706d8241065386ae10531f9d75437025) ...

Yeah, I had to change it there because it was comparing the wrong argument, thus the in-and-out of math mode wasn't working.  I see you already found the problem (not in the `D` grabber as I had thought).  Thanks!